### ISSUE TITLE
【求助】我推理出来的结果好像不太对

### ISSUE THREAD

**Issue by Amo5** [VamosC/CapHuman#4]
我使用的是你推荐的测试命令：
python inference.py --ckpt ckpts/caphuman.ckpt --vae_ckpt ckpts/vae-ft-mse-840000-ema-pruned.ckpt --model models/cldm_v15.yaml --sd_ckpt ckpts/Realistic_Vision_V3.0.ckpt --input_image examples/input_images/196251.png --pose_image examples/pose_images/pose1.png --prompt "a photo of a man wearing a suit in front of Space Needle"

我的结果：input_image | pose_image | out_img
![image](https://github.com/VamosC/CapHuman/assets/41940277/186a9d66-4284-4331-b8a0-80d48a849c2e)
看上去有两个问题：
1.Input_img的ID并没有保持
2.pose_img的姿态也没有得到体现。
是我哪里用错了吗？

**Comment by VamosC** [comment]
你好， 请问可以提供一些终端输出以及文件目录等更多的信息嘛

**Comment by Amo5** [comment]
> 你好， 请问可以提供一些终端输出以及文件目录等更多的信息嘛
**终端输出：**
logging improved.
Args: Namespace(ckpt='ckpts/caphuman.ckpt', control_ckpt=None, controlnet_strength=0.0, input_image='examples/input_images/196251.png', model='models/cldm_v15.yaml', modes='position,pose', n_prompt='', output_image='examples/output_images/out1.png', pose_image='examples/pose_images/pose1.png', prompt='a photo of a man wearing a suit in front of Space Needle', sd_ckpt='ckpts/Realistic_Vision_V3.0.ckpt', seed=12345, vae_ckpt='ckpts/vae-ft-mse-840000-ema-pruned.ckpt')
creating the FLAME Decoder
trained model found. load data/deca_model.tar
CapHuman: Running in eps-prediction mode
Setting up MemoryEfficientCrossAttention. Query dim is 320, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 320, context_dim is 768 and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 320, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 320, context_dim is 768 and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 640, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 640, context_dim is 768 and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 640, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 640, context_dim is 768 and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1280, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1280, context_dim is 768 and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1280, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1280, context_dim is 768 and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1280, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1280, context_dim is 768 and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1280, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1280, context_dim is 768 and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1280, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1280, context_dim is 768 and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1280, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1280, context_dim is 768 and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 640, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 640, context_dim is 768 and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 640, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 640, context_dim is 768 and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 640, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 640, context_dim is 768 and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 320, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 320, context_dim is 768 and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 320, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 320, context_dim is 768 and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 320, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 320, context_dim is 768 and using 8 heads.
DiffusionWrapper has 859.52 M params.
making attention of type 'vanilla-xformers' with 512 in_channels
building MemoryEfficientAttnBlock with 512 in_channels...
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
making attention of type 'vanilla-xformers' with 512 in_channels
building MemoryEfficientAttnBlock with 512 in_channels...
Setting up MemoryEfficientCrossAttention. Query dim is 320, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 320, context_dim is 768 and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 320, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 320, context_dim is 768 and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 640, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 640, context_dim is 768 and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 640, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 640, context_dim is 768 and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1280, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1280, context_dim is 768 and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1280, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1280, context_dim is 768 and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1280, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 1280, context_dim is 768 and using 8 heads.
Loaded model config from [models/cldm_v15.yaml]
Loaded state_dict from [ckpts/caphuman.ckpt]
Loaded state_dict from [ckpts/Realistic_Vision_V3.0.ckpt]
Loaded state_dict from [ckpts/vae-ft-mse-840000-ema-pruned.ckpt]
positive prompt = a photo of a man wearing a suit in front of Space Needle
negative prompt = 
not use meanshape
drop_control_cond_t: 0.0
Data shape for DDIM sampling is (1, 4, 64, 64), eta 0.0
Running DDIM Sampling with 20 timesteps

**文件目录：**
![image](https://github.com/VamosC/CapHuman/assets/41940277/b83c5290-5b28-46e3-a110-2505161f2501)

**Comment by VamosC** [comment]
@Amo5 从终端输出和文件目录来看都没问题，有可能是环境之类装的有问题么

**Comment by You-Cun** [comment]
我也遇到相同的问题，看了下中间结果，是deca得到的render、albedo和normal都是0，具体原因暂时不清楚

**Comment by gaoyixuan111** [comment]
I also got the same results during inference, with poor performance in pose control and identity retention. Has the reason for all zeros been identified? @VamosC @You-Cun

**Comment by VamosC** [comment]
Hi @Amo5 @You-Cun @gaoyixuan111,

I reproduce the result successfully and details are as follows. I hope these can help you.

# Environment
```bash
python -V
```
Python 3.7.16

```bash
cat /etc/os-release
```
PRETTY_NAME="Ubuntu 22.04.3 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.3 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```bash
pip list
```
[env.txt](https://github.com/user-attachments/files/17797072/env.txt)

# File structure
In CapHuman:
```
+- ckpts
+- |   face-parsing
+- |   |   79999_iter.pth
+- |   caphuman.ckpt
+- |   put checkpoints here.txt
+- |   Realistic_Vision_V3.0.ckpt
+- |   vae-ft-mse-840000-ema-pruned.ckpt
```

```
+- data
+- |   uv_face_eye_mask.png
+- |   fixed_displacement_256.npy
+- |   uv_face_mask.png
+- |   put face model data here.txt
+- |   texture_data_256.npy
+- |   landmark_embedding.npy
+- |   generic_model.pkl
+- |   deca_model.tar
+- |   FLAME_texture.npz
+- |   mean_texture.jpg
+- |   head_template.obj
```

In ~/.cache (/home/liangchao/.cache)
```
+- torch
+- |   hub
+- |   |   checkpoints
+- |   |   |   s3fd-619a316812.pth
+- |   |   |   resnet18-5c106cde.pth
+- |   |   |   resnet50-0676ba61.pth
+- |   |   |   2DFAN4-cd938726ad.zip
+- |   checkpoints
+- |   |   20180402-114759-vggface2.pt
```

```
+- huggingface                                                                                                                                                                       [107/1946]
+- |   hub
+- |   |   models--openai--clip-vit-large-patch14
+- |   |   |   .no_exist
+- |   |   |   |   32bd64288804d66eefd0ccbe215aa642df71cc41
+- |   |   |   |   |   added_tokens.json
+- |   |   |   snapshots
+- |   |   |   |   32bd64288804d66eefd0ccbe215aa642df71cc41
+- |   |   |   |   |   merges.txt
+- |   |   |   |   |   model.safetensors
+- |   |   |   |   |   vocab.json
+- |   |   |   |   |   tokenizer_config.json
+- |   |   |   |   |   config.json
+- |   |   |   |   |   tokenizer.json
+- |   |   |   |   |   special_tokens_map.json
+- |   |   |   |   |   preprocessor_config.json
+- |   |   |   refs
+- |   |   |   |   main
+- |   |   |   blobs
+- |   |   |   |   5a12a1eb250987a4eee0e3e7d7338c4b22724be1
+- |   |   |   |   9bfb42aa97dcd61e89f279ccaee988bccb4fabae
+- |   |   |   |   702bb12920b291cade3706cf215c1604d2255d93
+- |   |   |   |   76e821f1b6f0a9709293c3b6b51ed90980b3166b
+- |   |   |   |   a2bf730a0c7debf160f7a6b50b3aaf3703e7e88ac73de7a314903141db026dcb
+- |   |   |   |   2c19f6666e0e163c7954df66cb901353fcad088e
+- |   |   |   |   580c79c6862f31d1f9bd08dd1a415ba0d0502cd9
+- |   |   |   |   4297ea6a8d2bae1fea8f48b45e257814dcb11f69
```

Note that clip-vit-large-patch14 is automatically downloaded from huggingface and placed in ~/.cache/huggingface by default (refer to #6 if you need to download it manually). And some checkpoints related to 3D face reconstruction (DECA) and ID feature extraction (FaceNet - InceptionResnetV1) are automatically downloaded as well.

You will find more details in the output of the terminal shown below:
```
···
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 107M/107M [00:12<00:00, 9.18MB/s]
Downloading: "https://download.pytorch.org/models/resnet18-5c106cde.pth" to /home/liangchao/.cache/torch/hub/checkpoints/resnet18-5c106cde.pth
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 44.7M/44.7M [00:04<00:00, 10.3MB/s]
···
Downloading vocab.json: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 961k/961k [00:00<00:00, 4.18MB/s]
Downloading merges.txt: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 525k/525k [00:00<00:00, 37.4MB/s]
Downloading (…)cial_tokens_map.json: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 389/389 [00:00<00:00, 217kB/s]
Downloading tokenizer_config.json: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 905/905 [00:00<00:00, 819kB/s]
Downloading config.json: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4.52k/4.52k [00:00<00:00, 806kB/s]
Downloading model.safetensors: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1.71G/1.71G [00:15<00:00, 110MB/s]
···
Downloading (…)rocessor_config.json: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 316/316 [00:00<00:00, 82.5kB/s]
Downloading tokenizer.json: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2.22M/2.22M [00:00<00:00, 8.65MB/s]
···
Downloading: "https://www.adrianbulat.com/downloads/python-fan/s3fd-619a316812.pth" to /home/liangchao/.cache/torch/hub/checkpoints/s3fd-619a316812.pth
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 85.7M/85.7M [01:39<00:00, 900kB/s]
Downloading: "https://www.adrianbulat.com/downloads/python-fan/2DFAN4-cd938726ad.zip" to /home/liangchao/.cache/torch/hub/checkpoints/2DFAN4-cd938726ad.zip
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 91.9M/91.9M [06:05<00:00, 263kB/s]
···
```

# Command

```python
python inference.py --ckpt ckpts/caphuman.ckpt --vae_ckpt ckpts/vae-ft-mse-840000-ema-pruned.ckpt --model models/cldm_v15.yaml --sd_ckpt ckpts/Realistic_Vision_V3.0.ckpt --input_image examples/input_images/196251.png --pose_image examples/pose_images/pose1.png --prompt "a photo of a man wearing a suit in front of Space Needle" --output_image out-debug.png
```

# Output of the terminal
[log.txt](https://github.com/user-attachments/files/17797023/log.txt)

# Generation result
![out-debug](https://github.com/user-attachments/assets/48dcbee2-7f46-4519-b67b-84f4e44e2ae7)
